### PR TITLE
Feat/user stop progress

### LINKED ISSUE #38: Create POST/PUT a UserStopProgress
We need to create/update the progress of a stop for an user

Endpoint: POST/PUT `users/:id/stops/:id/progress`

### PR DESCRIPTION
**Pull request by kevinccbsg**
This PR fixes #38 

I use the upsert method to update the stop progress or create the record in case it does not exist.

I included a few tests with authentication and without.

One thing missing in this PR that I will address in a new one is how to handle if the user apart from being authenticated has access to that resource or not.

Probably I will work in middleware to solve that. That can be done in a new ticket because it affects other endpoints that require authentication.

### CODE REVIEW

File: server/routes/v1/stops.ts [hunk 1]
```diff
@@ -28,6 +31,47 @@ const initRouter = ({ controller, validators }: Dependencies) => {
     }
   });
 
+  /**
+   * POST /api/v1/stops/{id}/progress
```
Comment (SabrinaFZ) [inline server/routes/v1/stops.ts:35]: If it is updating why not using `PUT` instead of `POST` ?

File: server/routes/v1/stops.ts [hunk 2]
```diff
@@ -28,6 +31,47 @@ const initRouter = ({ controller, validators }: Dependencies) => {
...
+   * @summary Update user's stop progress
+   * @tags Stops
+   * @param {string} id.path - Stop id
+   * @param {StopProgressRequest} request.body.required - Progress payload
+   * @return {StopProgressResponse} 200 - List of comments from a stop
```
Comment (SabrinaFZ) [inline server/routes/v1/stops.ts:40]: Should return better 201 - Updated user stop progress

File: server/routes/v1/stops.ts [hunk 3]
```diff
@@ -28,6 +31,47 @@ const initRouter = ({ controller, validators }: Dependencies) => {
...
+   */
+  router.post(
+    '/:id/progress',
+    checkJwt,
+    validateRequest(),
+    async (req: IGetUserAuthInfoRequest, res, next) => {
+      try {
+        const { auth } = req;
+        const stopId = req.params.id;
+        if (!auth) throw unauthorizedError('Unauthorized');
```
Comment (SabrinaFZ) [inline server/routes/v1/stops.ts:52]: Maybe in another PR, but I'm starting to think that we can avoid DRY by checking the auth in an middleware 🤔

**Review by SabrinaFZ — CHANGES_REQUESTED** [review]
Some comments 👀

File: server/routes/v1/stops.ts (hunk 3 above)
Comment (kevinccbsg) [reply server/routes/v1/stops.ts:52]: yes in a new PR we could work on this and role validation.

**Review by SabrinaFZ — APPROVED** [review]
🚀🎃❤

**Pull request merged**